src/detection_linux.cpp

```diff
@@ -241,6 +232,14 @@ static void cbWork(uv_work_t *req) {
 			udev_device_unref(dev);
 		}
 	}
+
+	udev_monitor_unref(mon);
```

Comment:
Small thing but we should add a comment here to explain what's happening


```suggestion
	// After the loop stops running, clean up all of our references and close gracefully
	udev_monitor_unref(mon);
```